test/unit/test_utils.py

```diff
@@ -292,3 +301,23 @@ def test_mydumper_dump_processor_ignores_database_files_from_metadata_database()
         result4 = processor.process_line("-- metadata 0")
         assert result4 == "-- metadata 0"
         assert processor.gtid == _GTID
+
+
+def test_failed_migration_generates_file():
+    migration_error_path = Path("/", "tmp", "error.json")
+    server_uri = "mysql://user:password@invalid:3306"
+    migration = MySQLMigration(
+        source_uri=server_uri,
+        target_uri=server_uri,
+        target_master_uri=server_uri,
+        privilege_check_user="root@%",
+        migration_error_path=migration_error_path
+    )
+    with raises(OperationalError):
+        migration.start(migration_method=MySQLMigrateMethod.dump, seconds_behind_master=0)
+
+    with open(migration_error_path, encoding='utf-8') as file:
+        error: MysqlMigrationError = json.loads(file.read(), object_hook=lambda d: MysqlMigrationError(**d))
+        assert error.error_type == "pymysql.err.OperationalError"
+        assert error.error_msg is not None
+        assert error.error_date is not None
```

Comment:
Would be good to have stricter checks here.

Like error_date checked against a regex to make sure the timestamp is in isoformat with tz information. (e.g. `[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:-[0-9]{2}:-[0-9]{2}Z`)